api4/cloud.go

```diff
@@ -49,6 +50,29 @@ func (api *API) InitCloud() {
 
 	// POST /api/v4/cloud/webhook
 	api.BaseRoutes.Cloud.Handle("/webhook", api.CloudAPIKeyRequired(handleCWSWebhook)).Methods("POST")
+
+	api.BaseRoutes.Cloud.Handle("/notify-admin-to-upgrade", api.APISessionRequired(handleNotifyAdminToUpgrade)).Methods("POST")
+}
+
+var notifyLock sync.Mutex
```

Comment:
3/5 I'd recommend placing the lock in the app layer. My thinking:
* A lock is a bottleneck; it should be as short as possible. Currently, it would lock and block other requests even if it is a bad request that would never be processed and never need to read or write the notify admin system data.
* No other code in this project locks in the api layer; they are placed at the lower layers in to limit the bottleneck. Other projects I've worked on are similar.
* Keeping the lock limited to the piece of code that can cause races makes it much clearer why the lock exists.